server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java

```diff
@@ -881,14 +885,17 @@ private void publishAddedToSubmissionListPosts(CandidateOpportunity opp) {
         autoPostAddedToSubList.setContent("The candidate " + candidateNameAndNumber +
                 " is a prospect for the job '" + opp.getJobOpp().getName() +"'.");
 
-        // AUTO CHAT TO PROSPECT CHAT
-        JobChat prospectChat = jobChatService.getOrCreateJobChat(JobChatType.CandidateProspect, null,
-                null, candidate);
-        // Create the chat post
-        ChatPost prospectChatPost = chatPostService.createPost(
+        // Note that we don't post to candidates until they get past the prospect stage
+        if (opp.getStage() != CandidateOpportunityStage.prospect) {
```

Comment:
@camerojo I don't understand this condition check: the method `publishAddedToSubmissionListPosts()` is only called when the candidate is added to a submission list, at which point the stage is _always_ Prospect — so aren't lines 888-898 redundant?

Reply:
You are right - but I think that the publishAddedToSubmissionListPosts is poorly named. 

In the JavaDoc for that method it says
"Publish posts for after a Candidate Opportunity is created (e.g. Added to submission list)"

Note the "eg". The method is only called from the generic createOrUpdateCandidateOpportunity method. And that method allows for the possibility of a candidate opportunity being created with any initial stage (through the supplied CandidateOpportunityParams) - not just prospect. 

You are probably right that possibly right now candidate opportunities are only created by adding to a submission list - and the stage defaults to prospect. That wasn't the case in the past, and may not always be the case in the future. The name of publishAddedToSubmissionListPosts making assumptions about how the code got to this point, when it is and should in fact be a completely generic method (like createOrUpdateCandidateOpportunity from where it is called).

Does that make sense? @samschlicht

Reply:
Thanks, @camerojo, yes it does — although now the condition check assumes there's no chance an opp could be created at a stage _before_ Prospect (when we would also not want a post to be sent). 

I know in practice that will almost certainly be the case — it's just potentially a little confusing without some doc.

I know it's not quite so simple but what we would ideally have is `if (opp.getStage() > CandidateOpportunityStage.prospect) { `!

We're due a discussion on the issue that I picked up in the same area, so perhaps that's an opportunity to add some clarifying doc. Don't get me wrong, I don't think this is critical.